mediaengine.go

```diff
@@ -14,51 +14,52 @@ func RegisterCodec(codec *RTCRtpCodec) {
 	DefaultMediaEngine.RegisterCodec(codec)
 }
 
-// TODO: Phase out DefaultPayloadTypes in favor or dynamic assignment in 96-127 range
-
-// PayloadTypes for the default codecs
-const (
-	DefaultPayloadTypeOpus = 111
-	DefaultPayloadTypeVP8  = 96
-	DefaultPayloadTypeVP9  = 98
-	DefaultPayloadTypeH264 = 100
-)
-
 // RegisterDefaultCodecs is a helper that registers the default codecs supported by pions-webrtc
 func RegisterDefaultCodecs() {
-	RegisterCodec(NewRTCRtpOpusCodec(DefaultPayloadTypeOpus, 48000, 2))
-	RegisterCodec(NewRTCRtpVP8Codec(DefaultPayloadTypeVP8, 90000))
-	RegisterCodec(NewRTCRtpH264Codec(DefaultPayloadTypeH264, 90000))
-	RegisterCodec(NewRTCRtpVP9Codec(DefaultPayloadTypeVP9, 90000))
+	RegisterCodec(NewRTCRtpOpusCodec(48000, 2))
+	RegisterCodec(NewRTCRtpVP8Codec(90000))
+	RegisterCodec(NewRTCRtpH264Codec(90000))
+	RegisterCodec(NewRTCRtpVP9Codec(90000))
 }
 
 // DefaultMediaEngine is the default MediaEngine used by RTCPeerConnections
 var DefaultMediaEngine = NewMediaEngine()
 
 // NewMediaEngine creates a new MediaEngine
 func NewMediaEngine() *MediaEngine {
-	return &MediaEngine{}
+	return &MediaEngine{nextPayloadType: 96}
 }
 
 // MediaEngine defines the codecs supported by a RTCPeerConnection
 type MediaEngine struct {
 	codecs []*RTCRtpCodec
+
+	// store the next payload type used for dynamic assignment
+	nextPayloadType uint8
 }
 
 // RegisterCodec registers a codec to a media engine
 func (m *MediaEngine) RegisterCodec(codec *RTCRtpCodec) uint8 {
-	// TODO: generate PayloadType if not set
-	m.codecs = append(m.codecs, codec)
-	return codec.PayloadType
-}
 
-func (m *MediaEngine) getCodec(payloadType uint8) (*RTCRtpCodec, error) {
-	for _, codec := range m.codecs {
-		if codec.PayloadType == payloadType {
-			return codec, nil
+	var payloadType uint8
+
+	// check for existing payload type
+	for _, c := range m.codecs {
+		if c.Name == codec.Name {
```

Comment:
Is it sufficient to identify the codec by name only?

Reply:
The payload type defines a specific codec, including all its properties. Looking up a codec based on its properties can be done like in ``getCodecSDP``.